ERCS/erc-7715.md

```diff
@@ -0,0 +1,466 @@
+- `permissionsContext` is an opaque string that identifies the session. It MAY be the `context` as defined in [ERC-7679](./eip-7679.md) and [ERC-7710](./eip-7710.md).  See “Rationale” for details.
+
+If the request is malformed or the wallet is unable/unwilling to issue permissions, wallet MUST return an error with a code as defined in [ERC-1193](./eip-1193.md).
+
+`wallet_issuePermissions` response example:
+
+```tsx
+{
+  grantedPermissions: [
+    {
+      type: 'erc20_spending_limit',
+      data: {
+        erc20Address:'0x...',
+        limit: '0x...',
+      },
+    },
+    {
+      type: 'gas_limit',
+      data: {
+        limit: '0x...',
+      },
+      required: false
+    }
+  ],
+  
+  expiry: 1577840461,
+    
+  permissionsContext: '0x...',
+}
+```
+
+### Signer & Permission Types
+
+In this ERC, we specify a list of signers and permissions that we expect to be commonly used.
+
+This ERC does not specify an exhaustive list of signer or permission types, since we expect more signer/permission types to be developed as wallets get more advanced.  A signer/permission type is valid as long as both the DApp and the wallet are willing to support it.
+
+However, if two signers or two permissions share the same type name, a DApp could request with one type of signer/permission while the wallet grants another.  Therefore, it’s important that no two signers or two permissions share the same type.  Therefore, new signer or permission types should be specified in ERCs, either in this ERC as an amendment or in another ERC.
+
+#### Signers
+
+```tsx
+// A signer representing a single key.
+// `id` is a DID and can therefore represent both Secp256k1 or Secp256r1 keys, among other key types.
+type KeySigner = {
+  type: 'key',
+  data: {
+    id: string,
+  }
+}
+
+// A signer representing a multisig signer.
+// Each element of `ids` is a DID.
+type MultiKeySigner = {
+  type: 'keys',
+  data: {
+    ids: string[],
+  }
+}
+
+// An account that can be granted with permissions as in ERC-7710.
+type AccountSigner = {
+  type: 'account',
+  data: {
+    id: `0x${string}`,
+  }
+}
+```
+
+### Permissions Types
+
+```tsx
+// Native token spending limit, e.g. ETH on Ethereum
+type NativeTokenLimitPermission = {
+  type: 'native-token-limit',
+  data: {
+    amount: "0x...",  // wei in hex
+  }
+}
+
+// ERC20 spending limit
+type ERC20LimitPermission = {
+  type: 'erc20-limit',
+  data: {
+    erc20Address: "0x..."  // ERC20 token address
+    amount: "0x...",  // wei in hex
+  }
+}
+
+// Gas spending limit
+type GasLimitPermission = {
+  type: 'gas-limit',
+  data: {
+    amount: "0x...",  // wei in hex
+  }
+}
+
+// Allowlist of contract calls
+type ContractCallPermission = {
+  type: 'contract-call',
+  data: {
+    // TODO
+  }
+}
```

Comment:
What is the use-case for this?

Reply:
Limiting the call to a single smart contract method. For example, this permissions are valid only for using `purhcaseNft` method on smart contract x with the amount y. Its just an example of how permissions might look like.